File: src/main/java/net/sf/rails/game/round/Activity.java
package net.sf.rails.game.round;

import net.sf.rails.game.RailsAbstractItem;
import net.sf.rails.game.state.BooleanState;
import rails.game.action.PossibleAction;
import rails.game.action.PossibleActions;

public abstract class Activity extends RailsAbstractItem {
   
    private final BooleanState enabled = BooleanState.create(this, "enabled");
    
    protected Activity(RoundNG parent, String id) {
        super(parent, id);
    }

    public void setEnabled(boolean enabled) {
        this.enabled.set(enabled);
    }

    public boolean isEnabled() {
        return enabled.value();
    }
    
    /**
     * create actions and add them to the possibleActions object
     */
    public abstract void createActions(Actor actor, PossibleActions actions);
    
    /**
     * checks if the conditions of the actions are fullfilled
     */
    public abstract boolean isActionExecutable(PossibleAction action);
    
    /**
     * executes the action
     */
    public abstract void executeAction(PossibleAction action);
    
    /**
     * reports action execution
     */
    public abstract void reportExecution(PossibleAction action);
    
}

File: dotnet/RailsLib.Net/Net/Game/GameRound/Activity.cs
using GameLib.Rails.Game.Action;
using GameLib.Net.Game.State;
using System;


namespace GameLib.Net.Game.GameRound
{
    abstract public class Activity : RailsAbstractItem
    {
        private readonly BooleanState enabled; 


    protected Activity(RoundNG parent, String id) : base(parent, id)
        {
            enabled = BooleanState.Create(this, "enabled");
        }

        public void SetEnabled(bool enabled)
        {
            this.enabled.Set(enabled);
        }

        public bool IsEnabled
        {
            get
            {
                return enabled.Value;
            }
        }

        /**
         * create actions and add them to the possibleActions object
         */
        public abstract void CreateActions(IActor actor, PossibleActions actions);

        /**
         * checks if the conditions of the actions are fulfilled
         */
        public abstract bool IsActionExecutable(PossibleAction action);

        /**
         * executes the action
         */
        public abstract void ExecuteAction(PossibleAction action);

        /**
         * reports action execution
         */
        public abstract void ReportExecution(PossibleAction action);
    }
}
